docs/proposals/auditing.md

```diff
@@ -0,0 +1,260 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+**WORK-IN-PROCESS -- UNFINISHED -- WORK-IN-PROGRESS**
+
+## Abstract
+
+This proposal aims at extending the auditing log capabilities of the apiserver.
+
+## Motivation and Goals
+
+With https://github.com/kubernetes/kubernetes/pull/27087 basic audit logging is added to Kubernetes. It basically implements `access.log` like http handler based logging of all requests in the apiserver API. It does not do deeper inspection of the API calls or of their payloads. Moreover, it has no specific knowledge of the API objects which are modified. Hence, the log output does not answer the question how API objects actually change.
+
+The log output format of https://github.com/kubernetes/kubernetes/pull/27087 is fixed. It is text based, unstructured (e.g. non-JSON) data which must be parsed to be usable in any advanced external system used to analyse audit logs.
+
+The log output format does not follow any public standard like e.g. https://www.dmtf.org/standards/cadf.
+
+With this proposal we describe how the auditing functionality can be extended in order:
+
+- to allow multiple output formats, e.g. access.log style or structured JSON output
+- to allow deep payload inspection to allow
+  - either real differential JSON output (which field of an object have changed from which value to which value)
+  - or full object output of the new state (and optionally the old state)
```

Comment:
the wording "can be extended" is important here. Not all of this will be
implemented in a first/default audit logger.

On Thu, Jul 28, 2016 at 2:42 PM, Maciej Szulik <notifications@github.com>
wrote:

> In docs/proposals/auditing.md
> <https://github.com/kubernetes/kubernetes/pull/29443#discussion_r72613950>
> :
>
> > +This proposal aims at extending the auditing log capabilities of the apiserver.
> > +
> > +## Motivation and Goals
> > +
> > +With https://github.com/kubernetes/kubernetes/pull/27087 basic audit logging is added to Kubernetes. It basically implements `access.log` like http handler based logging of all requests in the apiserver API. It does not do deeper inspection of the API calls or of their payloads. Moreover, it has no specific knowledge of the API objects which are modified. Hence, the log output does not answer the question how API objects actually change.
> > +
> > +The log output format of https://github.com/kubernetes/kubernetes/pull/27087 is fixed. It is text based, unstructured (e.g. non-JSON) data which must be parsed to be usable in any advanced external system used to analyse audit logs.
> > +
> > +The log output format does not follow any public standard like e.g. https://www.dmtf.org/standards/cadf.
> > +
> > +With this proposal we describe how the auditing functionality can be extended in order:
> > +
> > +- to allow multiple output formats, e.g. access.log style or structured JSON output
> > +- to allow deep payload inspection to allow
> > +  - either real differential JSON output (which field of an object have changed from which value to which value)
> > +  - or full object output of the new state (and optionally the old state)
>
> I thought we've discussed only the latter case, where you output both and
> external system can do that diff.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29443/files/29439ba6ccc8651bb2129d1bb30214630a38c7e8#r72613950>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAskC5AguRY2NUydBXiDOJ7k1rJuxucWks5qaKOYgaJpZM4JSliE>
> .
>